ledfx/sendspin/stream.py

```
@@ -449,6 +531,16 @@ def close(self):
                 self._loop.call_soon_threadsafe(self._loop.stop)
             self._thread.join(timeout=2.0)
 
+        # Clean up pyFLAC decoder once the stream thread has fully stopped.
+        if self._flac_decoder is not None:
+            try:
+                self._flac_decoder.finish()
+            except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)